odoo/addons/base/models/ir_attachment.py

```diff
@@ -466,9 +501,6 @@ def read(self, fields=None, load='_classic_read'):
 
     def write(self, vals):
         self.check('write', values=vals)
-        # remove computed field depending of datas
```

Comment:
It is currently not possible to cleanly overwrite the `_inverse_datas` method. This is because in `write` the two fields `file_size` and `checksum` are automatically set to False. This makes it impossible to call write of super from the overwritten `_inverse_datas` method, as these fields will be deleted in any case. Both currently available solutions are not very clean. Either call super from `_inverse_datas` and then delete the unnecessarily data that has been created. Or the more performant method, but for the maintainability of the code problematic, directly call `models.Model.write(attach.sudo(), vals)` instead of `super(IrAttachment, attach.sudo()).write(vals)`.

After both fields are already readonly it should be possible to remove this check.